powerdnsadmin/routes/domain.py

```diff
@@ -131,6 +134,47 @@ def domain(domain_name):
                            ttl_options=ttl_options)
 
 
+@domain_bp.route('/remove', methods=['GET', 'POST'])
+@login_required
+@can_remove_domain
+def remove():
+    if request.method == 'POST':
+        current_app.logger.debug(request.form)
+        domain_name = request.form['domainid']
```

Comment:
The page is rendered only with domains you have access to, but during the deletion process there is no check to see if the domainid sent is a domain you own, which means, that any user can delete any domain by changing the form body

Reply:
Hello,

thanks.
Added a Check with new commit